src/main/scala/org/constellation/util/Rewards.scala

```diff
@@ -0,0 +1,72 @@
+package org.constellation.util
+
+object Rewards {
+  val roundingError = 0.000000000001
+
+  val epochOne = 438000
+  val epochTwo = 876000
+  val epochThree = 1314000
+  val epochFour = 1752000
+
+  val epochOneRewards = 0.68493150684//10k/mo, 328.767123287/day
+  val epochTwoRewards = 0.34246575342
+  val epochThreeRewards = 0.17123287671
+  val epochFourRewards = 0.08561643835
+
+  /*
+  Partitioning of address space, light nodes have smaller basis that full. Normalizes rewards based on node size
+   */
+  val partitonChart = Map[String, Set[String]]()
+
+  /*
+  Should come from reputation service
+   */
+  val transitiveReputationMatrix = Map[String, Map[String, Double]]()
+  val neighborhoodReputationMatrix = Map[String, Double]()
+
+  /*
+  snapshots - 10yrs @ ~20/hr
+   */
+  val rewardsPool = 1752000
+
+  def validatorRewards(
+                        curShapshot: Int,
+                       transitiveReputationMatrix: Map[String, Map[String, Double]],
+                       neighborhoodReputationMatrix: Map[String, Double],
+                       partitonChart: Map[String, Set[String]]
+                      )= {
+    val trustEntropyMap = shannonEntropy(transitiveReputationMatrix, neighborhoodReputationMatrix)
+    val distro = rewardDistribution(partitonChart, trustEntropyMap)
+    distro.mapValues(_ * rewardForEpoch(curShapshot))
+  }
+
+  def rewardForEpoch(curShapshot: Int) = curShapshot match {
+    case num if num >= 0 && num < epochOne  => epochOneRewards
+    case num if num >= 438000 && num < epochTwo => epochTwoRewards
+    case num if num >= 876000 && num < epochThree => epochThreeRewards
+    case num if num >= 1314000 && num < epochFour => epochFourRewards
+    case _ => 0d
+  }
+
+  def shannonEntropy(
+                      transitiveReputationMatrix: Map[String, Map[String, Double]],
+                      neighborhoodReputationMatrix: Map[String, Double]
+                    ) = {
+    val weightedTransitiveReputation = transitiveReputationMatrix.map { case (key, view) =>
+      val neighborView = view.map{ case (neighbor, score) => neighborhoodReputationMatrix(key) * score }.sum
+        (key, neighborView)
+    }
+    weightedTransitiveReputation.mapValues{ trust => - trust * math.log(trust)/math.log(2) }
+  }
+
+  def rewardDistribution(partitonChart: Map[String, Set[String]], trustEntropyMap: Map[String, Double]) = {
+    val totalSpace = partitonChart.values.map(_.size).max
+    val contributions = partitonChart.mapValues( partiton => partiton.size / totalSpace )
+    val weightedEntropy = contributions.map{ case (address, partitonSize) =>
```

Comment:
If I read this correctly, `partitonSize` is a `partitonSize` to the biggest contributor - is that correct? Not sure if I grok the `totalSpace` name here.

In the line below, could you elaborate why 1-entropy line? This is, if I'm not mistaken, per contributor something of the form 1-t*log(t)? But the second term t*log(t) isn't particularly normalized to anything, right? Not sure what this does, do you maybe happen to have a reference? 
Here, is it the case that the highest contributer gains the highest rewards.

If it could help - as I don't know who you're collaborating over those things - let me know if you want to make an elaborative chat over this. 
People will eventually hack into this and tweak if around, so I'd like to know if I can understand it.